src/primitives.rs

```diff
@@ -1,10 +1,167 @@
 //! Contains comparison primitives used to build up the rest of the library.
 
+use core::{
+    cmp::Ordering,
+    convert::{From, TryFrom},
+};
+
+// TODO: turn primitives.rs into
+// primitives/
+//   mod.rs
+//   types.rs for basic types
+//   traits.rs? or maybe per trait
+
+/// Represents a matching streak of characters between two strings.
+///
+/// See [find_longest_match] for details.
+#[derive(PartialEq, Eq, Copy, Clone, Debug)]
+pub struct MatchingStreak {
+    /// The index into the first (typically shorter) string where the streak
+    /// begins.
+    pub idx1: usize,
+    /// The index into the second (typically longer) string where the streak
+    /// begins.
+    pub idx2: usize,
+    /// The size of the matching character streak.
+    pub size: usize,
+}
+
+/// Represents an integer score of 0-100, inclusive.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Debug, Clone, Copy)]
+pub struct Score {
+    score: u8,
+}
+
+impl Score {
+    const MAX: u8 = 100;
+    pub fn new(score: u8) -> Option<Self> {
+        if score <= Self::MAX {
+            Some(Score { score })
+        } else {
+            None
+        }
+    }
+    pub fn score(&self) -> u8 {
+        self.score
+    }
+}
+
+impl TryFrom<u8> for Score {
+    // TODO: pick a better 'error' type...
```

Comment:
This is exactly the comment I leave whenever I'm too lazy to define an Enum and derive `thiserror::Error` 😆

Reply:
yea! I honestly don't know what, if any, error ecosystem we should plug into. Thoughts?

Reply:
The wisdom for a while has been "anyhow" for binaries, "thiserror" for libraries. The macros from thiserror are simply and non-magic enough that I feel comfortable using them normally.